app/src/features/User/UserRoute.tsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+import { CircularProgress } from "@material-ui/core";
+import { Redirect, Route, RouteProps } from "react-router";
+
+import { useApiUserRetrieveQuery } from "services/api/endpoints";
+
+const UserRoute = ({ ...routeProps }: RouteProps): JSX.Element => {
+  const { isLoading, data: user } = useApiUserRetrieveQuery({});
+  const isAuthenticated = Boolean(user);
+
+  if (isLoading) return <CircularProgress />;
+  if (isAuthenticated) return <Route {...routeProps} />;
+
+  return <Redirect to="/" />;
```

Comment:
Why not redirect to `OIDC_LOGIN_URL` ?

Reply:
I don't see why not! @simonvadee When the user is not yet logged in, is it fine we redirect the user automatically to the login page of the IDP ?